radis/tools/slit.py

```diff
@@ -500,11 +500,14 @@ def convolve_with_slit(
     waveunit="",
 ):
     """Convolves spectrum (w,I) with instrumental slit function (w_slit, I_slit)
-    Returns a convolved spectrum on a valid range.
+
+        Returns a convolved spectrum on a valid range. 
+
+        Check Notes section below for more details about its implementation.
 
         This function is called directly from a :py:class:`~radis.spectrum.spectrum.Spectrum`
         object with the :py:meth:`~radis.spectrum.spectrum.Spectrum.apply_slit` method.
-
+    
 
     .. warning::
```

Comment:
@gh4ag  note the existing warning here, which also [appears in the docs](https://radis.readthedocs.io/en/latest/source/radis.tools.slit.html#radis.tools.slit.convolve_with_slit).   It's not in [apply_slit](https://radis.readthedocs.io/en/latest/source/radis.spectrum.spectrum.html#radis.spectrum.spectrum.Spectrum.apply_slit, though. 

The fact it's not visible enough makes me wonder if the additional information you wrote should be added in the Notes of the function docstring , or elsewhere, maybe in the [Spectrum object guide](https://radis.readthedocs.io/en/latest/spectrum/spectrum.html#apply-instrumental-slit-function) ?

Reply:
Regarding get_FWHM(), I saw that the interpolation was not implemented yet, but for the slit function it is a must have.

The slit function that we provide to convolve_with_slit is made of not many points, and it follows the spirit of this functions since we ask users to cut the input slit function to its minimum spectral range.

I dit a test on this slit function and I got
[slit_fig.pdf](https://github.com/radis/radis/files/6629252/slit_fig.pdf)

Initially (FWHM): 0.2343906011052468
With get_FWHM: 0.16641365720090562

A gaussian fit yields something very close to 0.2389 nm. The main difference comes from interpolation.

Reply:
What's the reference for "initial = 0.234" ? Your slit file seems to have a resolution of ~0.07 nm so 0.166 and 0.234 nm are all within the error bar. 

You can use [plot_slit()](https://radis.readthedocs.io/en/latest/auto_examples/plot_slit_function.html#sphx-glr-auto-examples-plot-slit-function-py), it's a handy function to plot the slit & show the different FWHM (uses get_FWHM internally). 

Implementation wise : 

In convolve_with_slit we cannot afford to fit a gaussian all the time ; many slit functions aren't Gaussian (see example above) ; and it can be a bit costly although performance is key in fitting procedures . 

I'd suggest a very minimal test (compare w_slit_range to w_spectrum ?) first to determine if there can be a problem ; if so, then switch to a warning /error where performance doesn't matter anymore and the goal is to be be as informative as possible for the user : in this condition, you can afford fitting a Gaussian, or whatever seems best for you.

Reply:
I disagree with you estimation. By hand, I find 0.251 nm.

Reply:
"cannot afford to fit a gaussian" -> I would not even recommend it

Reply:
I modified the get_FWHM to work better: see 
b9abfe1

Reply:
Could you raise another issue for this ? I would like to use the output of this PR as soon as possible for my work :)